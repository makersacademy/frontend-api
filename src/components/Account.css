.ui.card {
  margin: auto !important;
  width: 100% !important;
  height: 100% !important;
}

div.signout.extra.content a {
  color: #4183c4 !important;
}