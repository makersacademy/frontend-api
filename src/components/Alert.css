.ui.message {
  width: 100% !important;
  position: absolute !important;
  display: none;
  bottom: 0;
  left: 0;
  opacity: 0;
  transition: all 2s ease-in-out !important;
}

