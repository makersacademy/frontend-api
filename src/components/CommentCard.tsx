import { CommentData } from "../../types/apiData";
import { ReactComponent as Tick } from "../assets/golden_tick.svg";

type Props = {
  data: CommentData;
};

export const CommentCard = ({ data }: Props) => {
  return (
    <div
      key={data.id}
      className="border flex gap-4 items-center rounded-xl p-4"
    >
      <div className="relative w-14 xl:w-20">
        <img
          data-cy="main-user-icon"
          src={`https://robohash.org/${data.id}`}
          alt=""
          className="object-cover border rounded-full"
        />
      </div>
      <div>
        <div className="flex xl:gap-1 xl:items-center flex-col xl:flex-row">
          <div className="flex gap-1">
            <h3 className={`text-sm xl:text-lg capitalize font-bold `}>
              {data.username}
            </h3>
            <Tick className="w-4 xl:w-5" />
          </div>
          <h4 className="text-secondary font-light text-sm xl:text-base">
            @{data.username}
          </h4>
        </div>
        <p>{data.content}</p>
      </div>
    </div>
  );
};
