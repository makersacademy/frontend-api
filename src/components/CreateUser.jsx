import { useState } from "react";
import axios from "axios";

const CreateUser = ({ onChangePage }) => {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");

  async function createUser(username, password) {
    const response = await axios.post(
      "https://chitter-backend-api-v2.herokuapp.com/users",
      {
        user: { handle: username, password: password },
      }
    );
    console.log(response);
    return response;
  }

  const onCreateFormSubmit = (event) => {
    event.preventDefault();
    console.log(username, password);
    createUser(username, password).then(
      (response) => response.status === "200" && onChangePage("Home")
    );
    // onChangePage("Home");
  };
  return (
    <div className="w-full max-w-xs">
      <form
        onSubmit={onCreateFormSubmit}
        className="bg-white shadow-md rounded px-8 pt-6 pb-8 mb-4"
      >
        <div className="mb-4">
          <label
            className="block text-gray-700 text-sm font-bold mb-2"
            htmlFor="username"
          >
            Username
          </label>
          <input
            className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
            id="username"
            type="text"
            placeholder="Username"
            value={username}
            onChange={(e) => setUsername(e.target.value)}
          />
        </div>
        <div className="mb-6">
          <label
            className="block text-gray-700 text-sm font-bold mb-2"
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
            id="password"
            type="password"
            placeholder="******************"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
        </div>
        <div className="flex items-center justify-between">
          <button
            className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
            type="submit"
            data-test="submit-btn"
          >
            Create User
          </button>
        </div>
      </form>
    </div>
  );
};

export default CreateUser;
