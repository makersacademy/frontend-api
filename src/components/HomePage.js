import React from 'react'
import Feed from './Feed'
import PeepForm from './PeepForm'

export class HomePage extends React.Component {

  render() {
    return (
      <div className="container">
        <h1>Chitter</h1>
        <PeepForm />
        <Feed />
      </div>

    )
  }
}


export default HomePage
