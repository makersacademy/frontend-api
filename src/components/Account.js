import React from "react";
import './Account.css'

const Account = ({ session, setSession }) => {

  const signOut = () => {
    localStorage.removeItem('user')
    setSession(false)
  }

  return (
    <div className="profile">
      <div className="ui card">
        <div className="image">
          <img src={session.avatar} alt='my beautiful pic'/>
        </div>
        <div className="content">
          <a href="#" className="header">{session.handle}</a>
          <div className="meta">
            <span className="date">Joined in 2019</span>
          </div>
          <div className="description">
            <div>Art Director - Los Angeles</div>
            <div>UCLA '15</div>
          </div>
        </div>
        <div className="extra content">
          <a>
            <i className="user icon"></i>
            22 Friends
          </a>
        </div>
        <div className="signout extra content">
          <a className="blue" href="#" onClick={signOut}>Sign Out</a>
        </div>
    </div>
  </div>
  )
}

export default Account