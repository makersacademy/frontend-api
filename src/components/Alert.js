import React, { useEffect } from "react";
import './Alert.css'

const Alert = ({ alert, setAlert }) => {
  
  useEffect(() => {

    if (alert) {
      const flash = document.querySelector('.flashAlert')
      if (flash) {
        flash.style.display = 'block'
        flash.style.opacity = 1;
      }
  
      let fadeTimeout = setTimeout(() => {
        if (flash) {
          flash.style.opacity = 0;
        }
      }, 2000)
  
      let showTimeout = setTimeout(() => {
        if (flash) {
          flash.style.display = 'none';
        }
      }, 4000)
  
      return () => {
        setAlert(false)
        const flash = document.querySelector('.flashAlert')
        clearTimeout(fadeTimeout)
        clearTimeout(showTimeout)
        if (flash) {
          flash.style.display = 'none'
        }
      }
    }

  }, [alert, setAlert])

    return (
      <div className={`ui ${alert.type} message flashAlert`} >
        <div className="header">
          {alert.message}
        </div>
      </div>
    )
 
}

export default Alert