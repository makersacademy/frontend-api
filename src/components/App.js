import React, { useState, useEffect } from "react";
import Menu from './Menu'
import LogIn from  './LogIn'
import Feed from './Feed'
import Peep from './Peep'
import Alert from './Alert' 

const App = () => {

  const [option, setOption] = useState('Feed');
  const [session, setSession] = useState(false)
  const [alert, setAlert] = useState(false)
  const [redirect, setRedirect] = useState(false)


  useEffect(() => {
    const user = localStorage.getItem('user')
    if (user) {
      setSession(JSON.parse(user))
    }
  }, [])

  return (
    <div className="app"> 
      <div className="ui top-bar">
        <Menu onSetOption={setOption} session={session} redirect={redirect} setRedirect={setRedirect} />
      </div>
      <div className='ui container content-field'>
        {
        option === 'LogIn' ? <LogIn setSession={setSession} session={session} setAlert={setAlert} />  :
        option === 'Feed' ? <Feed /> :
        option === 'Peep' ? <Peep redirect={setRedirect}/>
        : ''
        }
      <Alert alert={alert} setAlert={setAlert} />  
      </div>
    </div>
  )
}

export default App

