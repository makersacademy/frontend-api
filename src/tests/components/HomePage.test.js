import React from 'react';
import { shallow } from 'enzyme';
import { HomePage } from '../../components/HomePage'

const wrapper = shallow(<HomePage />)

test('should render HomePage correctly', () => {
  expect(wrapper).toMatchSnapshot();
})
