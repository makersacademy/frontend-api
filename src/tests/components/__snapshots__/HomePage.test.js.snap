// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HomePage correctly 1`] = `
<div
  className="container"
>
  <h1>
    Chitter
  </h1>
  <Connect(PeepForm) />
  <Connect(Feed) />
</div>
`;
